langchain/src/chains/combine_docs_chain.ts

```diff
@@ -154,31 +156,40 @@ export class MapReduceDocumentsChain
 
     let currentDocs = docs as Document[];
 
-    for (let i = 0; i < this.maxIterations; i += 1) {
```

Comment:
Would probably be easier to just do `this.maxIterations = Math.min(fields.maxIterations ?? this.maxIterations, SANITY_CHECK_MAX_ITERATION_COUNT)` in the constructor?